frame/staking/src/pallet/mod.rs

```diff
@@ -1593,6 +1586,12 @@ pub mod pallet {
 			//   threshold bond required.
 			//
 			// Otherwise, if caller is the same as the controller, this is just like `chill`.
+
+			if Nominators::<T>::contains_key(&stash) || Nominators::<T>::get(&stash).is_none() {
```

Comment:
Shouldn't these be an AND, not an OR? If `Nominators` contains the key but is Some (and thus decodable) we need to move on to check the other pre-conditions for chilling

Reply:
totally correct, almost f-ed up real bad there.